definitions/tools/add_strelka_gt.cwl

```diff
@@ -55,6 +55,27 @@ requirements:
                             $id++;
                         }
 
+                        # parse GT, handling unparseable case with stderr
+                        my $parsed_n_gt = parse_gt($n_gt_str, \%ids);
+                        if ($n_gt_info eq "ref") {
+                            $n_gt = "0/0";
```

Comment:
Since this case doesn't depend on parsing at all, maybe we should skip calling `parse_gt` if we're ending up here.

Reply:
I had that initially and I brought it outside mostly to avoid the extra nesting, especially since the strings are usually so short. I don't mind switching if others have a preference though

I'm guessing perl doesn't have the assignment expression thing python added in 3.8?
https://www.python.org/dev/peps/pep-0572/
I'd very much like to just have a single case that says "if parsed_n_gt is assigned with this call" which gives us both

Reply:
Are you asking if you can declare variables in the middle of expressions?  If so, yeah--you could do something like this if you wanted:

```perl
if (defined( my $parsed_n_gt = parse_gt(...) )) {
  say ("Got $parsed_n_gt as the value");
}
```
It'll be scoped to the conditional as if you did something like:
```perl
{
  my $parsed_n_gt = parse_gt(...);
  if (defined($parsed_n_gt)) {
     say ("Got $parsed_n_gt as the value");
  }
}
```
which is also to say you should be able to then use the value inside the blocks following `elsif` or `else` if needed, but not beyond.